scripts/repo_manager.py

```diff
@@ -102,6 +102,12 @@ class cgra_pnr(Repo):
     directory = "cgra_pnr"
 
     def install(self):
+        # submodule
+        run("git submodule update --init --recursive", cwd=repo.directory)
+        # use modern C++ to compile
+        os.environ["CC"] = "/usr/bin/gcc-7"
+        os.environ["CXX"] = "/usr/bin/g++-7"
```

Comment:
Just to be safe, can we store the old values of `CC` and `CXX` and then restore them after this install process is done? Or is there other code that depends on these env vars being set?

Alternatively, you could change the make line (112) to something like `run("CC=/usr/bin/gcc-7 CXX=/usr/bin/g++-7 make")` so it only sets the env vars for that command (that's unless you need those env vars for the pip commands). This is mainly to avoid any issues in the future with people assuming `CC` and `CXX` and not realizing that it may have been changed somewhere else in the script